src/main/java/net/dv8tion/jda/api/interactions/commands/CommandInteractionPayload.java

```diff
@@ -0,0 +1,273 @@
+    Command.Type getCommandType();
+
+    /**
+     * The command name.
+     * <br>This can be useful for abstractions.
+     *
+     * <p>Note that commands can have these following structures:
+     * <ul>
+     *     <li>{@code /name subcommandGroup subcommandName}</li>
+     *     <li>{@code /name subcommandName}</li>
+     *     <li>{@code /name}</li>
+     * </ul>
+     *
+     * You can use {@link #getCommandPath()} to simplify your checks.
+     *
+     * @return The command name
+     */
+    @Nonnull
+    String getName();
+
+    /**
+     * The subcommand name.
+     * <br>This can be useful for abstractions.
+     *
+     * <p>Note that commands can have these following structures:
+     * <ul>
+     *     <li>{@code /name subcommandGroup subcommandName}</li>
+     *     <li>{@code /name subcommandName}</li>
+     *     <li>{@code /name}</li>
+     * </ul>
+     *
+     * You can use {@link #getCommandPath()} to simplify your checks.
+     *
+     * @return The subcommand name, or null if this is not a subcommand
+     */
+    @Nullable
+    String getSubcommandName();
+
+    /**
+     * The subcommand group name.
+     * <br>This can be useful for abstractions.
+     *
+     * <p>Note that commands can have these following structures:
+     * <ul>
+     *     <li>{@code /name subcommandGroup subcommandName}</li>
+     *     <li>{@code /name subcommandName}</li>
+     *     <li>{@code /name}</li>
+     * </ul>
+     *
+     * You can use {@link #getCommandPath()} to simplify your checks.
+     *
+     * @return The subcommand group name, or null if this is not a subcommand group
+     */
+    @Nullable
+    String getSubcommandGroup();
+
+    /**
+     * Combination of {@link #getName()}, {@link #getSubcommandGroup()}, and {@link #getSubcommandName()}.
+     * <br>This will format the command into a path such as {@code mod/mute} where {@code mod} would be the {@link #getName()} and {@code mute} the {@link #getSubcommandName()}.
+     *
+     * <p>Examples:
+     * <ul>
+     *     <li>{@code /mod ban -> "mod/ban"}</li>
+     *     <li>{@code /admin config owner -> "admin/config/owner"}</li>
+     *     <li>{@code /ban -> "ban"}</li>
+     * </ul>
+     *
+     * @return The command path
+     */
+    @Nonnull
+    default String getCommandPath()
+    {
+        StringBuilder builder = new StringBuilder(getName());
+        if (getSubcommandGroup() != null)
+            builder.append('/').append(getSubcommandGroup());
+        if (getSubcommandName() != null)
+            builder.append('/').append(getSubcommandName());
+        return builder.toString();
+    }
+
+    /**
+     * Gets the display string for this command.
+     * <br>This is similar to the string you see when clicking the interaction name in the client.
+     *
+     * <p>Example return for an echo command: {@code /say echo phrase: Say this}
+     *
+     * @return The display string for this command
+     */
+    @Nonnull
+    default String getCommandString()
```

Comment:
It also makes sense for auto-complete

Reply:
Isn't AutoComplete a different type, so the return would have it return just the name?

Reply:
It still provides the entire command that is currently being input by the user

Reply:
I mean the `getCommandType() != Command.Type.SLASH` escape at the beginning would prevent it working with AutoComplete. But I'm realizing that it is comparing Command.Type, not Interaction.Type

Reply:
So to "properly" represent this we'd need a setup like:
CommandInteractionPayload -> CommandInteraction ->BaseSlashCommandInteraction

BaseSlashCommandInteraction -> SlashCommandInteraction
BaseSlashCommandInteraction -> CommandAutoCompleteInteraction

But honestly that seems like too much for a _single_ method.

Reply:
That being said, the docs on this should probably mention that it will only return the `getName` for non-slash-command style commands and then we'd be golden.